src/Rule/Nested.php

```diff
@@ -169,4 +168,36 @@ private function fetchOptions(array $rules): array
 
         return $result;
     }
+
+    private static function calculateErrorKey(Rules $aggregatedRule, string $valuePath, string $key): string
+    {
+        if (!self::canConcatenateErrorKey($aggregatedRule, $key)) {
+            return $valuePath;
+        }
+
+        return $valuePath . Result::ERROR_KEY_SEPARATOR . $key;
+    }
+
+    private static function canConcatenateErrorKey(Rules $aggregatedRule, string $key): bool
+    {
+        if (!self::isPositiveInteger($key)) {
+            return true;
+        }
+
+        $aggregatedRuleArr = $aggregatedRule->asArray();
+        if (ArrayHelper::getValueByPath($aggregatedRuleArr, '0.0') !== 'each') {
+            return false;
+        }
+
+        return !(ArrayHelper::getValueByPath($aggregatedRuleArr, '0.1.0') === 'nested');
+    }
+
+    private static function isPositiveInteger(string $str): bool
+    {
+        if (!preg_match('/^\d$/', $str)) {
```

Comment:
Is this check here to exclude scientific notation?

Reply:
It's needed to additionally exclude strings like `+5`. See https://stackoverflow.com/a/15277806/4323648 (2nd comment).

Reply:
:heavy_check_mark: